magenta/models/music_vae/lstm_utils.py

```diff
@@ -93,19 +93,48 @@ def _TFCanonicalNamePrefix(self, layer, is_fwd=True):
 
 
 def cudnn_lstm_state(lstm_cell_state):
-  """Convert tuple of LSTMCellStateTuples to CudnnLSTM format."""
+  """Convert tuple of LSTMStateTuples to CudnnLSTM format."""
   h = tf.stack([s.h for s in lstm_cell_state])
   c = tf.stack([s.c for s in lstm_cell_state])
   return (h, c)
 
 
-def get_final(time_major_sequence, sequence_length):
-  """Get the final item in a time-major sequence."""
-  final_index = tf.stack(
-      [tf.maximum(0, sequence_length - 1),
-       tf.range(sequence_length.shape[0])],
-      axis=1)
-  return tf.gather_nd(time_major_sequence, final_index)
+def reverse_cudnn_lstm_state(lstm_cell_state):
```

Comment:
The word "reverse" here sounds a little bit strange, maybe "extract_cudnn_lstm_state" or "cudnn_lstm_state_to_state_tuples" or something else?

Reply:
Done.